File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/DBG-1/cpp/dbg1.cpp
// DBG-1
#include <cstdint>
#include <cstdio>
#include <string>
#include <vector>

static const int64_t ITERATIONS = 300000000;
static const int64_t MODULUS = 1000000007;

struct Item {
    int64_t count;
    std::string name;
};

int64_t accumulate(const std::vector<Item>& items, int64_t iterations) {
    int64_t total = 0;
    for (int64_t i = 0; i < iterations; ++i) {
        for (const Item& item : items) {
            total = (total * 31 + item.count + (int64_t)item.name.size()) % MODULUS;
        }
    }
    return total;
}

int main() {
    std::vector<Item> items;
    items.push_back(Item{7, "alpha"});
    items.push_back(Item{11, "bravo"});
    items.push_back(Item{13, "charlie"});
    int64_t checksum = accumulate(items, ITERATIONS);
    printf("checksum=%lld\n", (long long)checksum);
    return 0;
}

File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/DBG-1/python/dbg1.py
# DBG-1: aggregate with named int + named string field, dynamic array of 3,
# a function that loops over the array, prints a checksum, runs >= 2s CPU.
ITERATIONS = 8000000
MODULUS = 1000000007


class Item:
    def __init__(self, count, name):
        self.count = count
        self.name = name


def accumulate(items, iterations):
    total = 0
    for _ in range(iterations):
        for item in items:
            total = (total * 31 + item.count + len(item.name)) % MODULUS
    return total


def main():
    items = []
    items.append(Item(7, "alpha"))
    items.append(Item(11, "bravo"))
    items.append(Item(13, "charlie"))
    checksum = accumulate(items, ITERATIONS)
    print("checksum={}".format(checksum))


main()
